corehq/ex-submodules/dimagi/utils/couch/__init__.py

```diff
@@ -377,8 +378,7 @@ def __init__(self, keys, fail_hard=False, timeout=60, block=True):
     def __enter__(self):
         try:
             for key in self.keys:
-                sep = "-" if "-" in key else "_"
-                name = "_".join(key.split(sep, 2)[:2])
+                name = "_".join(re.split(r"_|-", key)[:2])
```

Comment:
The `[:2]` seems specific to the `async_indicator_save-129879876323423` example. Are we that consistent about naming locks in a way that you really do always want the first two words of the key?

Reply:
I checked all of the things that call this and verified that the first two words constitute a unique name (at least I think they're mostly unique) while not including unnecessary runtime context (the most important part).

Reply:
Thinking about this more... To ensure future uses of `CriticalSection` do not accidentally cause a tag explosion in datadog, I may need to add a new required `name` argument to `CriticalSection` and revisit all of the call sights to explicitly pass in that value. I was trying to avoid that. 😞 

In the mean time, it is important to get this merged because the old code is generating thousands of new tags in datadog due to `
async_indicator_save-...` keys.